sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -3232,6 +3232,35 @@ def compile_and_reload(
             pipeline_func=pipeline, package_path=output_yaml)
         return components.load_component_from_file(output_yaml)
 
+class TestResourceConfig(unittest.TestCase):
+    def test_cpu_memory_optional(self):
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            predict_op = components.load_component_from_text("""
+      name: predict
+      inputs:
+      - {name: generate_explanation, type: Boolean, default: False}
+      implementation:
+        container:
+          image: gcr.io/my-project/my-image:tag
+          args:
+          - {inputValue: generate_explanation}
+      """)
```

Comment:
Can we use a v2 Python component with the `@dsl.component` decorator instead? There are some examples of this throughout this file. Here are the [docs](https://www.kubeflow.org/docs/components/pipelines/v2/author-a-pipeline/components/#1-lighweight-python-function-based-components).

The style used here is v1 components, which is a deprecated authoring approach.